packages/gguf/scripts/generate-llm.ts

```diff
@@ -0,0 +1,194 @@
+/**
+ * Script for generating llm.ts
+ * The source data is taken from llama.cpp
+ */
+
+import { writeFileSync } from "node:fs";
+
+const SOURCE_CPP_URL = "https://raw.githubusercontent.com/ggerganov/llama.cpp/master/llama.cpp";
+const DEST_FILE_PATH = "./src/transformer-llm.ts";
+const DEST_COMMON_SOURCE = `
+type Attention<TArchitecture extends string> =
+	& { [K in \`\${TArchitecture}.attention.head_count\`]: number }
+	& { [K in \`\${TArchitecture}.attention.head_count_kv\`]: number }
+	& { [K in \`\${TArchitecture}.attention.layer_norm_epsilon\`]: number }
+	& { [K in \`\${TArchitecture}.attention.layer_norm_rms_epsilon\`]: number }
+	& { [K in \`\${TArchitecture}.attention.alibi_bias_max\`]: number }
+	& { [K in \`\${TArchitecture}.attention.clip_kqv\`]: number }
+	& { [K in \`\${TArchitecture}.attention.use_norm\`]: number };
+
+type Rope<TArchitecture extends LLMArchitecture> =
+	& { [K in \`\${TArchitecture}.rope.dimension_count\`]: number }
+	& { [K in \`\${TArchitecture}.rope.freq_base\`]: number }
+	& { [K in \`\${TArchitecture}.rope.scale\`]: number }
+	& { [K in \`\${TArchitecture}.rope.scale_linear\`]: number };
+
+type MOE<TArchitecture extends LLMArchitecture> =
+	& { [K in \`\${TArchitecture}.expert_count\`]: number }
+	& { [K in \`\${TArchitecture}.expert_used_count\`]: number };
+
+export type TransformerLLMArchitecture = LLMArchitecture; // type alias
+export type TransformerLLMBase<TArchitecture extends LLMArchitecture> = ModelBase<TArchitecture>
+	& MOE<TArchitecture>
+	& Attention<TArchitecture>
+	& Rope<TArchitecture>;
+`;
+
+const KV_TYPE = {
+	LLM_KV_ATTENTION_LAYERNORM_RMS_EPS: "number",
+	LLM_KV_ATTENTION_LAYERNORM_EPS: "number",
+	LLM_KV_ATTENTION_CAUSAL: "boolean",
+	LLM_KV_TOKENIZER_TOKEN_TYPE_COUNT: "number",
+	LLM_KV_POOLING_TYPE: "number", // TODO: this is an enum
```

Comment:
is this todo being handled in this PR or follow-up PR ?

Reply:
handled in https://github.com/huggingface/huggingface.js/commit/6962b5143a70631f5a2736b53ba1ddbdc8d609f0